src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -8554,6 +8544,200 @@ public void ParamsArray_ThisModifier_02()
+                    diagnostics.Add(
+                        // (7,24): warning CS9100: Parameter 1 has params modifier in lambda but not in target delegate type.
+                        //         D d = (params  a) => { };
+                        Diagnostic(ErrorCode.WRN_ParamsArrayInLambdaOnly, "a").WithArguments("1"));
+                }
+            }
+
+            compilation.VerifyDiagnostics([.. diagnostics]);
+        }
+
+        [Fact]
+        public void TestScopedImplicitParameter_CSharp10()
+        {
+            var source = $$"""
+                using System;
+
+                class scoped { }
+
+                class C
+                {
+                    void M()
+                    {
+                        Action<scoped> f = (scoped a) => { };
+                    }
+                }
+                """;
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.Regular10).VerifyDiagnostics(
+                // (3,7): warning CS8981: The type name 'scoped' only contains lower-cased ascii characters. Such names may become reserved for the language.
+                // class scoped { }
+                Diagnostic(ErrorCode.WRN_LowerCaseTypeName, "scoped").WithArguments("scoped").WithLocation(3, 7));
+
+            var tree = compilation.SyntaxTrees[0];
+            var semanticModel = compilation.GetSemanticModel(tree);
+            var lambda = tree.GetRoot().DescendantNodes().OfType<ParenthesizedLambdaExpressionSyntax>().Single();
+
+            var symbol = (IMethodSymbol)semanticModel.GetSymbolInfo(lambda).Symbol;
+            Assert.Equal(ScopedKind.None, symbol.Parameters[0].ScopedKind);
+        }
+
+        [Fact]
+        public void TestScopedImplicitParameter_CSharp13()
+        {
+            var source = $$"""
+                using System;
+
+                class scoped { }
+
+                class C
+                {
+                    void M()
+                    {
+                        Action<scoped> f = (scoped a) => { };
+                    }
+                }
+                """;
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.Regular13).VerifyDiagnostics(
+                // (3,7): error CS9062: Types and aliases cannot be named 'scoped'.
+                // class scoped { }
+                Diagnostic(ErrorCode.ERR_ScopedTypeNameDisallowed, "scoped").WithLocation(3, 7));
+
+            var tree = compilation.SyntaxTrees[0];
+            var semanticModel = compilation.GetSemanticModel(tree);
+            var lambda = tree.GetRoot().DescendantNodes().OfType<ParenthesizedLambdaExpressionSyntax>().Single();
+
+            var symbol = (IMethodSymbol)semanticModel.GetSymbolInfo(lambda).Symbol;
+            Assert.Equal(ScopedKind.None, symbol.Parameters[0].ScopedKind);
+        }
+
+        [Fact]
+        public void TestScopedImplicitParameter()
+        {
+            var source = $$"""
+                class scoped { }
```

Comment:
>class scoped { } [](http://example.com/codeflow?start=16&length=16)

Consider using `class @scoped { }` to avoid the error.

Reply:
done.